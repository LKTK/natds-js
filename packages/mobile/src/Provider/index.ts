import { Provider as PaperProvider } from 'react-native-paper';

export class Provider extends PaperProvider{}
