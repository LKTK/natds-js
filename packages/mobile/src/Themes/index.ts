export { themes } from '@naturacosmeticos/natds-styles';
