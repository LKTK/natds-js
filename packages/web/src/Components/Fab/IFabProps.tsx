import {FabProps} from "@material-ui/core/Fab";

export type IFabColor = "primary" | "secondary" | "light";
export type IFabSize = "small" | "medium" | "large";
export type IFabVariant = "round" | "extended";

export interface IFabProps extends Pick<FabProps,
  | "children"
  | "disabled"
  | "disableFocusRipple"
  | "href"
  | "size"
  | "variant"> {

  /**
   * @optional
   * The color of the component
   */
  color ?: IFabColor;
}

export const colors: Record<IFabColor, IFabColor> = {
  primary: "primary",
  secondary: "secondary",
  light: "light",
};

export const defaultProps = {
  color: colors.primary,
  disabled: false,
};

export const sizes: Record<IFabSize, IFabSize> = {
  small: "small",
  medium: "medium",
  large: "large",
};

export const variants: Record<IFabVariant, IFabVariant> = {
  round: "round",
  extended: "extended",
};
