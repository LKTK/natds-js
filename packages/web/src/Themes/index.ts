import {
  themes as styleThemes,
} from "@naturacosmeticos/natds-styles";
import { parseTheme } from "./parseTheme";
import { ThemeBrandName } from "./ThemeBrandName";
import createThemesObject from "./createThemesObject";
import { IThemesWeb } from "./IThemesWeb";

export { IThemeWeb } from "./IThemeWeb";

export const themes = Object.keys(styleThemes).reduce((value: IThemesWeb<ThemeBrandName>, key: string) => {

  const result = value;

  result[key as ThemeBrandName].dark = parseTheme(styleThemes[key as ThemeBrandName].dark);
  result[key as ThemeBrandName].light = parseTheme(styleThemes[key as ThemeBrandName].light);

  return result;
}, createThemesObject());
