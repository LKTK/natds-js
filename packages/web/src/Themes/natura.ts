import { themes, tokens } from '@naturacosmeticos/natds-styles';
import { createMuiTheme } from '@material-ui/core';

const { spacing }  = tokens.spacing;
const { naturaLight } = themes;
const { colors, roundness } = naturaLight;

const natdsTheme = createMuiTheme({
  spacing,
  palette: {
    primary: {
      main: colors.primary
    },
    secondary: {
      main: colors.secondary
    },
    error: {
      main: colors.error
    },
    type: 'light',
  },
  shape: {
    borderRadius: roundness
  }
});

export default natdsTheme;
