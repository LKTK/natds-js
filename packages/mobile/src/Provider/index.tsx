import { Provider as PaperProvider } from 'react-native-paper';
import * as React from 'react';
import { natura } from '../Themes';
import { ITheme } from '@naturacosmeticos/natds-styles';

interface IProvider {
  children: React.ReactNode;
  theme?: ITheme;
}

const Provider: React.FunctionComponent<IProvider> = (props) => {
  const { theme } = props;
  return <PaperProvider {...props} theme={theme ? theme : natura} />;
};

export default Provider;
