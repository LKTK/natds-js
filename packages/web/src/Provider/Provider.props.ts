import * as React from "react";
import { IThemeWeb } from "../Themes";

export interface IProviderProps {

  /**
   * @required
   * @type node
   */
  children: React.ReactNode;

  /**
   * Defaults to Natura Light theme
   *
   * @optional
   * @type IThemeWeb
   */
  theme?: IThemeWeb;
}
