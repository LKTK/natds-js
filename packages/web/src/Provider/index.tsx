/**
 * @todo Review/re-enable `jsx-no-string-ref` tslint rule
 * @todo Review/re-enable `no-unused-expression` tslint rule
 * @todo Review/re-enable `use-isnan` tslint rule
 * @todo Review/re-enable `triple-equals` tslint rule
 */
import { MuiThemeProvider } from '@material-ui/core/styles';
import createMuiTheme, { ThemeOptions } from '@material-ui/core/styles/createMuiTheme';
import * as React from 'react';

import { IThemeWeb, themes } from '../Themes';

export interface IProvider {
  children: React.ReactNode;
  theme?: IThemeWeb;
}

export const Provider:React.FunctionComponent<IProvider> = (props:IProvider) => {

  /**
   * @todo fix(web): TS2322 - Type is not assignable to type 'ThemeOptions'. Types of some properties are incompatible.
   */
  // @ts-ignore
  const theme: ThemeOptions = props.theme ? { ...props.theme } : { ...themes.natura.light },
    newTheme = createMuiTheme(theme);


  return (
    <MuiThemeProvider {...props} theme={newTheme} />
  );
};
