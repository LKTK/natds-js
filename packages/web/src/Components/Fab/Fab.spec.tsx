import * as React from "react";
import * as renderer from "react-test-renderer";

import { Fab } from ".";
import { Provider } from "../../Provider";
import { themes } from "../../Themes";
import {IFabColor, IFabSize, IFabVariant} from "./IFabProps";

describe("Fab component", () => {
  describe("rendering", () => {
    const colors: IFabColor[] = [
      "primary",
      "secondary",
      "light",
    ];
    const sizes: IFabSize[] = [
      "large",
      "medium",
      "small",
    ];
    const variants: IFabVariant[] = [
      "round", "extended",
    ];

    test("it should render with no props", () => {
      const wrapper : renderer.ReactTestRenderer = renderer.create(<Fab>F</Fab>);

      expect(wrapper.toJSON()).toMatchSnapshot("No Props Fab");
    });

    test("it should render with Provider theme", () => {
      const wrapper : renderer.ReactTestRenderer = renderer.create(<Provider theme={themes.natura.dark}><Fab>F</Fab></Provider>);

      expect(wrapper.toJSON()).toMatchSnapshot("Provider Theme Fab");
    });

    test("it should render with disabled truthy", () => {
      const wrapper : renderer.ReactTestRenderer = renderer.create(<Fab disabled>F</Fab>);

      expect(wrapper.toJSON()).toMatchSnapshot("Disabled Fab");
    });

    test("it should render all colors", () => {
      colors.forEach((color: IFabColor) => {
        const wrapper : renderer.ReactTestRenderer = renderer.create(<Fab color={color}>F</Fab>);

        expect(wrapper.toJSON()).toMatchSnapshot(`Color ${color} Fab`);
      });
    });

    test("it should render all sizes", () => {
      sizes.forEach((size: IFabSize) => {
        const wrapper : renderer.ReactTestRenderer = renderer.create(<Fab size={size}>F</Fab>);

        expect(wrapper.toJSON()).toMatchSnapshot(`Size ${size} Fab`);
      });
    });

    test("it should render all variants", () => {
      variants.forEach((variant: IFabVariant) => {
        const wrapper : renderer.ReactTestRenderer = renderer.create(<Fab variant={variant}>F</Fab>);

        expect(wrapper.toJSON()).toMatchSnapshot(`Size ${variant} Fab`);
      });
    });
  });
});
