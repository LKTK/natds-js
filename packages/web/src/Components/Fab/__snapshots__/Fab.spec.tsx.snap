// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fab component rendering it should render all colors: Color light Fab 1`] = `
<button
  className="MuiButtonBase-root MuiFab-root makeStyles-root-16"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiFab-label"
  >
    F
  </span>
  <span
    className="MuiTouchRipple-root"
  />
</button>
`;

exports[`Fab component rendering it should render all colors: Color primary Fab 1`] = `
<button
  className="MuiButtonBase-root MuiFab-root makeStyles-root-10 MuiFab-primary makeStyles-primary-11"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiFab-label"
  >
    F
  </span>
  <span
    className="MuiTouchRipple-root"
  />
</button>
`;

exports[`Fab component rendering it should render all colors: Color secondary Fab 1`] = `
<button
  className="MuiButtonBase-root MuiFab-root makeStyles-root-13 MuiFab-secondary makeStyles-secondary-15"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiFab-label"
  >
    F
  </span>
</button>
`;

exports[`Fab component rendering it should render all sizes: Size large Fab 1`] = `
<button
  className="MuiButtonBase-root MuiFab-root makeStyles-root-19 MuiFab-primary makeStyles-primary-20"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiFab-label"
  >
    F
  </span>
</button>
`;

exports[`Fab component rendering it should render all sizes: Size medium Fab 1`] = `
<button
  className="MuiButtonBase-root MuiFab-root makeStyles-root-22 MuiFab-sizeMedium MuiFab-primary makeStyles-primary-23"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiFab-label"
  >
    F
  </span>
  <span
    className="MuiTouchRipple-root"
  />
</button>
`;

exports[`Fab component rendering it should render all sizes: Size small Fab 1`] = `
<button
  className="MuiButtonBase-root MuiFab-root makeStyles-root-25 MuiFab-sizeSmall MuiFab-primary makeStyles-primary-26"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiFab-label"
  >
    F
  </span>
</button>
`;

exports[`Fab component rendering it should render all variants: Size extended Fab 1`] = `
<button
  className="MuiButtonBase-root MuiFab-root makeStyles-root-31 MuiFab-extended MuiFab-primary makeStyles-primary-32"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiFab-label"
  >
    F
  </span>
</button>
`;

exports[`Fab component rendering it should render all variants: Size round Fab 1`] = `
<button
  className="MuiButtonBase-root MuiFab-root makeStyles-root-28 MuiFab-primary makeStyles-primary-29"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiFab-label"
  >
    F
  </span>
  <span
    className="MuiTouchRipple-root"
  />
</button>
`;

exports[`Fab component rendering it should render with Provider theme: Provider Theme Fab 1`] = `
<button
  className="MuiButtonBase-root MuiFab-root makeStyles-root-4 MuiFab-primary makeStyles-primary-5"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiFab-label"
  >
    F
  </span>
  <span
    className="MuiTouchRipple-root"
  />
</button>
`;

exports[`Fab component rendering it should render with disabled truthy: Disabled Fab 1`] = `
<button
  className="MuiButtonBase-root MuiFab-root makeStyles-root-7 Mui-disabled MuiFab-primary makeStyles-primary-8 Mui-disabled"
  disabled={true}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={-1}
  type="button"
>
  <span
    className="MuiFab-label"
  >
    F
  </span>
</button>
`;

exports[`Fab component rendering it should render with no props: No Props Fab 1`] = `
<button
  className="MuiButtonBase-root MuiFab-root makeStyles-root-1 MuiFab-primary makeStyles-primary-2"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiFab-label"
  >
    F
  </span>
</button>
`;
